plugins/account_history/plugin.cpp

```diff
@@ -135,242 +157,290 @@ if( options.count(name) ) { \
+        }
 
         return pimpl->database.with_weak_read_lock([&]() {
-            return pimpl->get_account_history(account, from, limit);
+            return pimpl->get_account_history(account, from, limit, dir);
         });
     }
 
     struct get_impacted_account_visitor final {
-        fc::flat_set<golos::chain::account_name_type>& impacted;
+        fc::flat_map<golos::chain::account_name_type, operation_direction_type>& impacted;
 
-        get_impacted_account_visitor(fc::flat_set<golos::chain::account_name_type>& impact)
+        get_impacted_account_visitor(fc::flat_map<golos::chain::account_name_type, operation_direction_type>& impact)
             : impacted(impact) {
         }
 
         using result_type = void;
 
         template<typename T>
         void operator()(const T& op) {
-            op.get_required_posting_authorities(impacted);
-            op.get_required_active_authorities(impacted);
-            op.get_required_owner_authorities(impacted);
+            fc::flat_set<golos::chain::account_name_type> impd;
+            op.get_required_posting_authorities(impd);
+            op.get_required_active_authorities(impd);
+            op.get_required_owner_authorities(impd);
+            for (auto i : impd) {
+                impacted.insert(make_pair(i, operation_direction_type::any));
+            }
         }
 
         void operator()(const account_create_operation& op) {
-            impacted.insert(op.new_account_name);
-            impacted.insert(op.creator);
+            impacted.insert(make_pair(op.new_account_name, operation_direction_type::receiver));
+            impacted.insert(make_pair(op.creator, operation_direction_type::sender));
         }
 
         void operator()(const account_create_with_delegation_operation& op) {
-            impacted.insert(op.new_account_name);
-            impacted.insert(op.creator);
+            impacted.insert(make_pair(op.new_account_name, operation_direction_type::receiver));
+            impacted.insert(make_pair(op.creator, operation_direction_type::sender));
         }
 
         void operator()(const account_update_operation& op) {
-            impacted.insert(op.account);
+            impacted.insert(make_pair(op.account, operation_direction_type::any));
         }
 
         void operator()(const account_metadata_operation& op) {
-            impacted.insert(op.account);
+            impacted.insert(make_pair(op.account, operation_direction_type::any));
         }
 
         void operator()(const comment_operation& op) {
-            impacted.insert(op.author);
-            if (op.parent_author.size()) {
-                impacted.insert(op.parent_author);
+            if (op.author == op.parent_author) {
+                impacted.insert(make_pair(op.author, operation_direction_type::any));
+            } else {
+                if (op.parent_author.size()) {
+                    impacted.insert(make_pair(op.author, operation_direction_type::receiver));
```

Comment:
why? comment author is always sender and parent_author is always receiver (excl. `any` case)